storage/storage_image.go

```diff
@@ -540,6 +576,39 @@ func (s *storageImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 	return reused, info, s.queueOrCommit(ctx, info, *options.LayerIndex)
 }
 
+// PutBlobPartial attempts to create a blob using the data that is already present at the destination storage.  stream is accessed
+// in a non-sequential way to retrieve the missing chunks.
+func (s *storageImageDestination) PutBlobPartial(ctx context.Context, stream types.ImageSourceSeekable, srcInfo types.BlobInfo, cache types.BlobInfoCache) (types.BlobInfo, error) {
+	fetcher := zstdFetcher{
+		stream:   stream,
+		ctx:      ctx,
+		blobInfo: srcInfo,
+	}
+
+	// While the PutBlobPartial() interface is generic, at this point only one format is supported (zstd:chunked), so  get a
+	// zstd:chunked differ and apply the diff using it.
+	differ, err := zstdchunked.MakeZstdChunkedDiffer(ctx, s.imageRef.transport.store, srcInfo.Size, srcInfo.Annotations, &fetcher)
```

Comment:
How is this actually going to work in the future with multiple formats? I guess based on `srcInfo.MediaType`, but how _exactly_?
- (Is that field even consistently filled in at this point?)
- `MediaType` is manifest-format-specific. So do we actually need some kind of mapping or reuse of data from `c/image/manifest` to turn `MediaType` into a known format? Is this something this function can handle itself, or something the copy pipeline must provide? I’m digging into the weeds here because this probably impacts the “what is `zstd:chunked`, is it a compression algorithm/variant/neither” discussion elsewhere.

Reply:
moved the logic to c/storage